show-geodesic-path-between-two-points/src/main/java/com/esri/arcgismaps/sample/showgeodesicpathbetweentwopoints/MainActivity.kt

```diff
@@ -0,0 +1,197 @@
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        // add all our marker graphics to the graphics overlay
+        graphicsOverlay.graphics.add(startingLocationMarkerGraphic)
+        graphicsOverlay.graphics.add(endLocationMarkerGraphic)
+        graphicsOverlay.graphics.add(geodesicPathGraphic)
+
+        lifecycleScope.launch {
+            // check if the map has loaded successfully
+            map.load().onSuccess {
+                // capture and collect when the user taps on the screen
+                mapView.onSingleTapConfirmed.collect { event ->
+                    event.mapPoint?.let { point -> onGeoViewTapped(point) }
+                }
+            }.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map", mapView)
+            }
+        }
+    }
+
+    /**
+     * Handles the SingleTapEvent by drawing the destination location marker
+     * and a geodesic path curve using GeometryEngine.densifyGeodetic
+     * and computes the distance using
+     * GeometryEngine.lengthGeodetic
+     */
+    private fun onGeoViewTapped(point: Point) {
+        // project the tapped point into the same spatial reference as source point
+        val destinationPoint = GeometryEngine.project(point, SpatialReference.wgs84())
+
+        // check if the destination point is within the map bounds
+        // isEmpty returns true if out of bounds
+        if (!destinationPoint.isEmpty) {
+            // update the end location marker location on map
+            endLocationMarkerGraphic.geometry = destinationPoint
+            // create a polyline between source and destination points
+            val polyline = Polyline(listOf(startingPoint, destinationPoint))
+            // generate a geodesic curve using the polyline
+            GeometryEngine.densifyGeodetic(
+                polyline,
+                1.0,
+                unitsOfMeasurement,
+                GeodeticCurveType.Geodesic
+                // only compute the distance if the returned curved path geometry is not null
+            )?.let { pathGeometry ->
+                // update the path graphic
+                geodesicPathGraphic.geometry = pathGeometry
+                // compute the path distance in kms
+                val distance =
+                    GeometryEngine.lengthGeodetic(
+                        pathGeometry,
+                        unitsOfMeasurement,
+                        GeodeticCurveType.Geodesic
+                    )
+                // display the distance result
+                infoTextView.text =
+                    getString(R.string.distance_info_text, distance.roundToInt())
+            }
+        }
+    }
+
+    /**
+     * Displays an error onscreen
+     */
+    private fun showError(message: String, view: View) {
+        Log.e(TAG, message)
+        Snackbar.make(view, message, Snackbar.LENGTH_SHORT).show()
+    }
```

Comment:
```suggestion
    private fun showError(message: String) {
        Log.e(TAG, message)
        Snackbar.make(mapView, message, Snackbar.LENGTH_SHORT).show()
    }
```
Can use `mapView` as the View here

Reply:
I believe the sample generator creates it with this signature.